Taken from linux-2.4.18-smallpatches.patch from kernel-2.4.21-27.0.4.EL.src.rpm
See: 
http://linux.bkbits.net:8080/linux-2.6/cset@412a4baaEebwtKg-X7sS2r5Mua6uGw
https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=133113
http://lkml.org/lkml/2004/7/29/332

diff -Naur linux-2.4.20.ori/fs/proc/base.c linux-2.4.20/fs/proc/base.c
--- linux-2.4.20.ori/fs/proc/base.c	2005-04-25 15:42:17.000000000 -0400
+++ linux-2.4.20/fs/proc/base.c	2005-04-25 16:01:40.000000000 -0400
@@ -185,8 +185,12 @@
 	int res = 0;
 	task_lock(task);
 	mm = task->mm;
-	if (mm)
-		atomic_inc(&mm->mm_users);
+	if (mm) {
+		if (mm->arg_end)
+			atomic_inc(&mm->mm_users);
+		else
+			mm = NULL;
+	}
 	task_unlock(task);
 	if (mm) {
 		int len = mm->arg_end - mm->arg_start;
